File: Q2.py
#Q2.#Create two vectors of type A=(a1,a2,a3) and B=(b1,b2,b3) with numbers of your choice (not random) in the code itself. Then Find A+B and A.B.


A = [10,0,1]
B = [-4,3,7]

dotproduct=0

print("The first vector =", A)
print("The second vector =", B)

#A+B
sum=0
for i,j in zip(A,B):
    sum=[i+j for i,j in zip(A,B)]
print ("A+B=", sum) 

#A.B
for i,j in zip(A,B):
    product =[i*j for i,j in zip(A,B)]
    dotproduct += i*j
print ("A.B=", dotproduct)


'''

The first vector = [10, 0, 1]                                                                                                                                                     
The second vector = [-4, 3, 7]                                                                                                                                                    
A+B= [6, 3, 8]                                                                                                                                                                     
A.B= -33                                                                                                                                                                            

'''

File: Q2.c
//#Create two vectors of type A=(a1,a2,a3) and B=(b1,b2,b3) with numbers of your choice (not random) in the code itself. Then Find A+B and A.B.


#include <stdio.h>
#include <stdlib.h>

int main()
{
	int A[3] = {10,0,1};

printf("Vector A = [");
    for (int i = 0; i < 3; i++)
    	printf("  %d  ", A[i]);
printf("]");

    int B[3] = {-4,3,7};
    
printf("\nVector B = [");
    for (int i = 0; i < 3; i++)
    	printf("  %d  ", B[i]);
printf("]");

	
	int sum[3], dotproduct=0;

	for (int i=0;i<3;i++)
    {
        sum[i] = A[i] + B[i];
    }

	printf("\nA + B = [");
	for (int i = 0; i < 3; i++)
		printf("  %d  ", sum[i]);
	printf("]");

    
    for (int i = 0; i < 3; i++)
	{
		dotproduct += (A[i] * B[i]);
	}

	printf("\nA.B= %d\n", dotproduct);

}


/*


Vector A = [  10    0    1  ]                                                                                                                                                                                              
Vector B = [  -4    3    7  ]                                                                                                                                                                                              
A + B = [  6    3    8  ]                                                                                                                                                                                                  
A.B= -33                                                                                                                                                                                                                   
         
*/